requests/script.js

```diff
@@ -1,16 +1,44 @@
 const API_BASE_URL = window.API_BASE_URL;
 const requestContainer = document.getElementById(REQUEST_CONTAINER_ID);
+const lastElementContainer = document.querySelector(LAST_ELEMENT_CONTAINER);
+
 const params = new URLSearchParams(window.location.search);
 const isDev = params.get('dev') === 'true';
 const loader = document.querySelector('.container__body__loader');
 const startLoading = () => loader.classList.remove('hidden');
 const stopLoading = () => loader.classList.add('hidden');
-let isDataLoading = false;
 let oooTabLink = document.getElementById(OOO_TAB_ID);
 let currentReqType = OOO_REQUEST_TYPE;
 let selected__tab__class = 'selected__tab';
 let statusValue = null;
 let sortByValue = null;
+let userDetails = [];
+let nextLink = '';
+let isDataLoading = false;
+
+function getUserDetails(id) {
+  return userDetails.find((user) => user.id === id);
+}
+
+const intersectionObserver = new IntersectionObserver(async (entries) => {
+  if (!nextLink) {
+    return;
+  }
+  if (entries[0].isIntersecting && !isDataLoading) {
+    await renderOooRequestCards({
+      state: statusValue,
+      sort: sortByValue,
+      next: nextLink,
+    });
+  }
+});
+
+const addIntersectionObserver = () => {
+  intersectionObserver.observe(lastElementContainer);
+};
+const removeIntersectionObserver = () => {
```

Comment:
where is this function called or used?

Reply:
We will used to stop observing. Currently it is not in used.